src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1162,7 +1162,12 @@ private bool IsAssigned(BoundExpression node, out int unassignedSlot)
             }
 
             Debug.Assert(unassignedSlot > 0);
-            return this.State.IsAssigned(unassignedSlot);
+            if (unassignedSlot > 0)
+            {
```

Comment:
If the if condition can be met, then the assertion can fail. Should the assertion be removed?

Reply:
> If the if condition can be met, then the assertion can fail. Should the assertion be removed?

We are still expecting the assert to not fail, even though we don't to crash if it is.

---
In reply to: [537702704](https://github.com/dotnet/roslyn/pull/49828#discussion_r537702704) [](ancestors = 537702704)